solvers/fast_int.py

```diff
@@ -0,0 +1,131 @@
+# - skipping import to speed up autocompletion in CLI.
+# - getting requirements info when all dependencies are not installed.
+with safe_import_context() as import_ctx:
+    from benchopt.stopping_criterion import SingleRunCriterion
+    import os
+    import numpy as np
+    from sklearn.preprocessing import StandardScaler
+    from benchmark_utils.config import MEMORY
+    from hyperalignment.fast_int import INT as HyperAlignment
+
+
+# The benchmark solvers must be named `Solver` and
+# inherit from `BaseSolver` for `benchopt` to work properly.
+class Solver(BaseSolver):
+    # Name to select the solver in the CLI and to display the results.
+    name = "FastINT"
+
+    # List of parameters for the solver. The benchmark will consider
+    # the cross product for each key in the dictionary.
+    # All parameters 'p' defined here are available as 'self.p'.
+    parameters = {}
+
+    # List of packages needed to run the solver. See the corresponding
+    # section in objective.py
+    install_cmd = "conda"
+    requirements = ["pip:hyperalignment", "joblib"]
+
+    stopping_criterion = SingleRunCriterion()
+
+    def set_objective(
+        self,
+        dict_alignment,
+        dict_decoding,
+        data_alignment_target,
+        data_decoding_target,
+        dict_labels,
+        target,
+        mask,
+    ):
+        # Define the information received by each solver from the objective.
+        # The arguments of this function are the results of the
+        # `Objective.get_objective`. This defines the benchmark's API for
+        # passing the objective to the solver.
+        # It is customizable for each benchmark.
+        self.dict_alignment = dict_alignment
+        self.dict_decoding = dict_decoding
+        self.data_alignment_target = data_alignment_target
+        self.data_decoding_target = data_decoding_target
+        self.dict_labels = dict_labels
+        self.target = target
+        self.mask = mask
+
+    def run(self, n_iter=None):
+        # This is the function that is called to evaluate the solver.
+        # It runs the algorithm for a given a number of iterations `n_iter`.
+        # You can also use a `tolerance` or a `callback`, as described in
+        # https://benchopt.github.io/performance_curves.html
+        X_train = []
+        y_train = []
+        X_test = []
+
+        fast_int_path = os.path.join(MEMORY, "fastint")
+        if not os.path.exists(fast_int_path):
+            os.makedirs(fast_int_path)
+
+        fast_int = HyperAlignment(n_jobs=5)
+
+        # Build alignment array
+        alignment_array = [
+            self.mask.transform(contrasts)
+            for _, contrasts in self.dict_alignment.items()
+        ]
+
+        alignment_array.append(self.mask.transform(self.data_alignment_target))
+        alignment_array = np.array(alignment_array)
+
+        # Compute the projected data into the common space (tunning matrices) for alignment data
```

Comment:
Typo
```suggestion
        # Compute the projected data into the common space (tuning matrices) for alignment data
```

Reply:
Realised halfway through my internship that it was not the correct spelling haha